src/v/raft/consensus.cc

```diff
@@ -618,6 +618,10 @@ ss::future<result<model::offset>> consensus::linearizable_barrier() {
     // store current commit index
     auto cfg = config();
     auto dirty_offset = _log->offsets().dirty_offset;
+    /**
+     * Flush log on leader, to make sure the _commited_index will be updated
+     */
+    co_await flush_log();
```

Comment:
couple of questions

* flush_log() doesn't update the _commit_index, right? should we call refresh_commit_index()?
* why is this needed? If _commit_index is stale, the user can issue a barrier again until _commit_index is at an offset that they want, couldn't convince myself this is needed.

Reply:
`flush_log()` is indeed not increasing committed index, but when the heartbeats are received with the flushed offsets updated then the index is incremented. 

In a situation where we used `acks=1` we want linerizable barrier to actually be linearizable in this case we should force committed offset update.

Reply:
ohh acks=1